packages/rollup-plugin-html/src/assets/utils.ts

```diff
@@ -143,3 +144,11 @@ export function getSourcePaths(node: Element) {
 export function findAssets(document: Document) {
   return findElements(document, isAsset);
 }
+
+// picomatch follows glob spec and requires "./" to be removed for the matcher to work
+// it is safe, because with or without it resolves to the same file
+// read more: https://github.com/micromatch/picomatch/issues/77
+const removeLeadingSlash = (str: string) => (str.startsWith('./') ? str.slice(2) : str);
+export function createAssetPicomatchMatcher(glob?: string | string[]) {
+  return picomatch(glob || [], { format: removeLeadingSlash });
+}
```

Comment:
this was pretty unexpected, but the proposed workaround from the picomatch maintainer seems to be good for us
https://github.com/micromatch/picomatch/issues/77

I added different variations in the test, with and without `./`, to be sure it works